src/fits.jl

```diff
@@ -0,0 +1,153 @@
+# helper functions
+
+# Gets the data from hdu
+function getdata(hdu::ImageHDU)
+    data = read(hdu)
+    d = ndims(data)
+    return permutedims(data, d:-1:1)
+end
+
+#------------------------------------------------------------------
+
+# FITS interface for basic reduction methods
+
+# code generated with codegen
+for func in (:flat_correct, :subtract_bias)
+    @eval $func(frame::ImageHDU, correction::AbstractArray; kwargs...) = $func(getdata(frame), correction; kwargs...)
+    @eval $func(frame::AbstractArray, correction::ImageHDU; kwargs...) = $func(frame, getdata(correction); kwargs...)
+    @eval $func(frame::ImageHDU, correction::ImageHDU; kwargs...) = $func(getdata(frame), correction; kwargs...)
+    @eval $func(frame::String, correction; hdu = 1, kwargs...) = $func(FITS(frame)[hdu], correction; kwargs...)
+    @eval $func(frame, correction::String; hdu = 1, kwargs...) = $func(frame, FITS(correction)[hdu]; kwargs...)
+    @eval begin
+        function $func(frame::String, correction::String; hdu = (1, 1), kwargs...)
+            hdus = hdu isa Integer ? (hdu, hdu) : hdu
+            return $func(FITS(frame)[hdus[1]], FITS(correction)[hdus[2]]; kwargs...)
+        end
+    end
+end
+
+for func in (:flat_correct!, :subtract_bias!)
+    @eval $func(frame::AbstractArray, correction::ImageHDU; kwargs...) = $func(frame, getdata(correction); kwargs...)
+    @eval $func(frame::AbstractArray, correction::String; hdu = 1, kwargs...) = $func(frame, FITS(correction)[hdu]; kwargs...)
+end
+
```

Comment:
I would add some developer comments here explaining the code gen. For instance, what is different between these for-loops and the above for-loops?